backend/persistent/datastores/DataWorld.js

```diff
@@ -0,0 +1,118 @@
+import fetch from 'node-fetch';
+import url from 'url';
+
+import Logger from '../../logger';
+
+function parseUrl(datasetUrl) {
+  const pathnameArray = url.parse(datasetUrl).pathname.split('/');
+  return {
+    owner: pathnameArray[1],
+    id: pathnameArray[2]
+  };
+}
+
+export function connect(connection) {
+  const { owner, id } = parseUrl(connection.url);
+  return new Promise((resolve, reject) => {
+    fetch(`https://api.data.world/v0/datasets/${owner}/${id}/`, {
+      method: 'GET',
+      headers: {
+        'Authorization': `Bearer ${connection.token}`,
+        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+        'User-Agent': `Falcon/Plotly - ${process.env.npm_package_version}`
+      }
+    })
+    .then(res => res.json())
+    .then(json => {
+      // json.code is defined only when there is an error
+      if (json.code) {
+        reject(json);
+      } else {
+        resolve();
+      }
+    })
+    .catch(err => {
+      Logger.log(err);
+      throw new Error(err);
+    });
+  });
+ }
+
+export function tables(connection) {
+  return new Promise((resolve, reject) => {
+    query('SELECT * FROM Tables', connection).then((res) => {
+      const allTables = res.rows.map((table) => {
+        return table[0];
+      });
+
+      resolve(allTables);
+    })
+    .catch(err => {
+      reject(err);
+    });
+  });
+}
+
+export function schemas(connection) {
+  return new Promise((resolve) => {
```

Comment:
No need to create a new Promise, just `return query(...` and replace `resolve({...})` by `return {...}`.